src/main/java/bio/terra/service/dataset/DatasetSummary.java

```diff
@@ -91,17 +97,50 @@ public DatasetSummary storage(List<? extends StorageResource<?, ?>> storage) {
         return this;
     }
 
-    public CloudRegion getStorageResourceRegion(CloudResource storageResource) {
-        return storage.stream()
-            .filter(resource -> resource.getCloudResource() == storageResource)
+    public List<BillingProfileModel> getBillingProfiles() {
+        return billingProfiles;
+    }
+
+    public DatasetSummary billingProfiles(List<BillingProfileModel> billingProfiles) {
+        this.billingProfiles = billingProfiles;
+        return this;
+    }
+
+    public BillingProfileModel getDefaultBillingProfile() {
+        return billingProfiles.stream()
+            .filter(b -> b.getId().equals(defaultProfileId))
             .findFirst()
-            .map(StorageResource::getRegion)
-            .orElseThrow(() -> new StorageResourceNotFoundException(
-                String.format("%s could not be found for %s ", storageResource, id)));
+            .orElseThrow();
+    }
+
+    public CloudRegion getStorageResourceRegion(CloudResource storageResource) {
+        return getCloudResourceAttribute(storageResource, StorageResource::getRegion);
     }
 
     public boolean datasetStorageContainsRegion(GoogleRegion region) {
         return storage.stream()
-                .anyMatch(sr -> sr.getRegion().equals(region));
+            .anyMatch(sr -> sr.getRegion().equals(region));
+    }
+
+    public CloudPlatform getStorageCloudPlatform() {
+        // A Dataset should not have both a bucket and a storage account at this point
+        return storage.stream().filter(s -> s.getCloudResource() == GoogleCloudResource.BUCKET ||
+            s.getCloudResource() == AzureCloudResource.STORAGE_ACCOUNT)
+            .findAny()
+            .map(s -> getCloudResourceAttribute(s.getCloudResource(), StorageResource::getCloudPlatform))
```

Comment:
I'm confused by this. You already have a `StorageResource` (`s`) whose `CloudResource` is `s.getCloudResource()`. You pass `s`'s cloud resource to a method then then finds another (?) `StorageResource` whose `CloudResource` is the same as `s`'s, and return its `CloudPlatform`. Is the one you find different than the current `s`? Why wouldn't it be correct to return `s.getCloudPlatform()` here directly instead?
```suggestion
            .map(StorageResource::getCloudPlatform))
```

Reply:
Whoop. An example of trying to reuse an existing method that doesn't _actually_ do what I want.